docs/src/snap/howto/install/install-offline.md

```diff
@@ -0,0 +1,374 @@
+
+# pull the image and tag
+ctr image pull "$FROM_REPOSITORY/$IMAGE"
+ctr image convert "$FROM_REPOSITORY/$IMAGE" "$TO_REPOSITORY/$IMAGE"
+```
+
+Finally, push the image (see `ctr image push --help` for a complete list of
+supported arguments):
+
+```bash
+# push image
+ctr image push "$TO_REPOSITORY/$IMAGE"
+# OR, if using HTTP and basic auth
+ctr image push "$TO_REPOSITORY/$IMAGE" --plain-http -u "$USER:$PASS"
+# OR, if using HTTPS and a custom CA 
+# (assuming CA certificate is at `/path/to/ca.crt`)
+ctr image push "$TO_REPOSITORY/$IMAGE" --ca /path/to/ca.crt
+```
+
+Make sure to repeat the steps above (pull, convert, push)
+for all the images that you need.
+
+##### Load images with docker
+
+On the machine with access to both registries, first install `docker`.
+For Ubuntu hosts, this can be done with:
+
+```bash
+sudo apt-get update
+sudo apt-get install docker.io
+```
+
+If needed, login to the private registry:
+
+```bash
+sudo docker login $TO_REGISTRY
+```
+
+Then pull, tag and push the image:
+
+```bash
+export IMAGE=library/nginx:latest
+export FROM_REPOSITORY=docker.io
+export TO_REPOSITORY=10.100.100.100:5000
+
+sudo docker pull "$FROM_REPOSITORY/$IMAGE"
+sudo docker tag "$FROM_REPOSITORY/$IMAGE" "$TO_REPOSITORY/$IMAGE"
+sudo docker push "$TO_REPOSITORY/$IMAGE"
+```
+
+Repeat the pull, tag and push steps for all required images.
+
+### Images Option C: Side-load images
+
+Image side-loading is the process of loading all required OCI images directly
+into the container runtime, so that they do not have to be fetched at runtime.
+Upon choosing this option, you need to create a bundle of all the OCI images
+that will be used by the cluster.
+
+#### Create a bundle
+
+From a running cluster you may import images with `ctr`.
+Ensure you grab all images from all nodes in the cluster.
+  
+If we have an OCI image called nginx.tar,
+we can load this to all the cluster nodes by running the following command
+on any of them:
+
+```bash
+sudo k8s ctr image import - < nginx.tar
+```
+<!-- TODO: I think ctr is already installed where? -->
+On success, the output will look like this:
+
+```bash
+unpacking docker.io/library/nginx:latest \
+(sha256:9c58d14962869bf1167bdef6a6a3922f607aa823196c392a1785f45cdc8c3451)...d
+```
+
+For all standard OCI images that you will use, from .tar archives.
+
+## Deploy Canonical Kubernetes
+
+Now that you have fulfilled all steps in preparation for your
+air gapped cluster, it is time to get it deployed.
+
+### Step 1: Install Canonical Kubernetes
+
+Copy the `k8s.snap`, `k8s.assert`, `core20.snap` and `core20.assert` files into
+the target node, then install with:
+
+```bash
+sudo snap ack core20.assert && sudo snap install ./core20.snap
+sudo snap ack k8s.assert && sudo snap install ./k8s.snap --classic
+```
+
+Repeat the above for all nodes of the cluster.
+
+### Step 2: Form Canonical Kubernetes cluster
+
+Now are ready to bootstrap the cluster by running:
+
+```bash
+sudo k8s bootstrap
```

Comment:
here, let's add `--address $IP`. we specify the address, therefore we can configure all Kubernetes services accordingly